conans/client/cmd/uploader.py

```diff
@@ -158,8 +158,31 @@ def _check_recipe_date(self, conan_ref, remote):
 
         if (remote_recipe_manifest != local_manifest and
                 remote_recipe_manifest.time > local_manifest.time):
+            self._print_manifest_information(remote_recipe_manifest, local_manifest, conan_ref, remote)
             raise ConanException("Remote recipe is newer than local recipe: "
                                  "\n Remote date: %s\n Local date: %s" %
                                  (remote_recipe_manifest.time, local_manifest.time))
 
         return remote_recipe_manifest
+
+    def _print_manifest_information(self, remote_recipe_manifest, local_manifest, conan_ref, remote):
+        try:
+            self._user_io.out.info("\n%s" % ("-"*40))
+            self._user_io.out.info("Remote manifest:")
+            self._user_io.out.info(remote_recipe_manifest)
+            self._user_io.out.info("Local manifest:")
+            self._user_io.out.info(local_manifest)
+            difference = remote_recipe_manifest.difference(local_manifest)
+            if "conanfile.py" in difference:
+                contents = tools.load(os.path.join(self._client_cache.export(conan_ref),
+                                                   "conanfile.py"))
+                endlines = "\\r\\n" if "\r\n" in contents else "\\n"
+                self._user_io.out.info("Local 'conanfile.py' using '%s' line-ends" % endlines)
```

Comment:
There can be mixed line endings in the same file (Visual Studio people knows about it, it is a common error 😩 )

Reply:
Sure. It is only a bit more information that can be totally useless in some cases but very useful sometimes.